File: python/ex-1.py

"""
Exercise - 1

1-Create employee data excel file with duplicate record  that contain the fallowing field
Emp id
Emp Name
Emp Salary
Write a python code to display duplicate employee record and count no of duplicate record available into a employee data file
"""


# using pandas library to parse csv file
import pandas

fileName = 'employee-det.csv'

df = pandas.read_csv(fileName)

# convert employee id coolumn to list
a = df['Employee id'].values.tolist()
# convert list to dictonary with key being employee id and value being total number of occurance of each key
result = dict((i, a.count(i)) for i in a)

totalDuplicates = 0;
duplicates = []

# iterate through the map
for keys in result.keys():
    # if number of occurance is more than one then the key is repeated more than one
    if result[keys] > 1:
        # store the duplicates and increase the total duplicates count
        duplicates.append(keys)
        totalDuplicates+=1

# print the result
print(result);
print('duplicates employee ids ',duplicates)
print('total duplicates ',totalDuplicates)

File: go/ex-1.go
package main

/**

Exercise - 1

1-Create employee data excel file with duplicate record  that contain the fallowing field
Emp id
Emp Name
Emp Salary
Write a python code to display duplicate employee record and count no of duplicate record available into a employee data file
**/

import (
	//"bufio"
	"encoding/csv"
	"io"
	"log"
	"os"
)

func main() {
	// Open the file
	csvfile, err := os.Open("employee-det.csv")
	if err != nil {
		log.Fatalln("Couldn't open the csv file", err)
	}
	r := csv.NewReader(csvfile)
	duplicates := map[string]int{}
	totalDuplicates := 0

	for {
		record, err := r.Read()

		if err == io.EOF {
			break
		}
		if err != nil {
			log.Fatal(err)
		}

		if _, ok := duplicates[record[0]]; ok {
			duplicates[record[0]] = duplicates[record[0]] + 1
			totalDuplicates++
		} else {
			duplicates[record[0]] = 1
		}
	}
	println("Total duplicates ", totalDuplicates)
	println("duplicates ", totalDuplicates)
}
